hedera-node/cli-clients/src/main/java/com/hedera/services/cli/sign/AccountBalanceSigningUtils.java

```diff
@@ -0,0 +1,124 @@
+import com.swirlds.platform.util.BootstrapUtils;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.security.InvalidKeyException;
+import java.security.KeyPair;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.SignatureException;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Utility class for signing account balance files
+ */
+public class AccountBalanceSigningUtils {
+
+    private static final Logger logger = LogManager.getLogger(AccountBalanceSigningUtils.class);
+    /**
+     * Hidden constructor
+     */
+    private AccountBalanceSigningUtils() {}
+
+
+
+    /**
+     * Sets up the constructable registry, and configures
+     * <p>
+     * Should be called before using stream utilities
+     * <p>
+     * Do <strong>NOT</strong> call this during standard node operation, as it will interfere with static settings
+     */
+    public static void initializeSystem() {
+        BootstrapUtils.setupConstructableRegistry();
+
+        // we don't want deserialization to fail based on any of these settings, not needed for services
+        //        SettingsCommon.maxTransactionCountPerEvent = Integer.MAX_VALUE;
+        //        SettingsCommon.maxTransactionBytesPerEvent = Integer.MAX_VALUE;
+        //        SettingsCommon.transactionMaxBytes = Integer.MAX_VALUE;
+        //        SettingsCommon.maxAddressSizeAllowed = Integer.MAX_VALUE;
+    }
+
+    /**
+     * Generates a signature file for the account balance file
+     *
+     * @param signatureFileDestination the full path where the signature file will be generated
+     * @param streamFileToSign         the stream file to be signed
+     * @param keyPair                  the keyPair used for signing
+     * @return true if the signature file was generated successfully, false otherwise
+     */
+    public static boolean signAccountBalanceFile(
+            @NonNull final Path signatureFileDestination,
+            @NonNull final Path streamFileToSign,
+            @NonNull final KeyPair keyPair) {
+
+        Objects.requireNonNull(signatureFileDestination, "signatureFileDestination must not be null");
+        Objects.requireNonNull(streamFileToSign, "streamFileToSign must not be null");
+        Objects.requireNonNull(keyPair, "keyPair must not be null");
+
+        try {
+            final Hash entireHash = computeEntireHash(streamFileToSign.toFile());
+            final byte[] fileHashByte = entireHash.getValue();
+            final byte[] signature = sign(fileHashByte, keyPair);
+
+            generateSigBalanceFile(signatureFileDestination.toFile(), signature, fileHashByte);
+
+            System.out.println("Generated signature file: " + signatureFileDestination);
```

Comment:
Why not use logger?

Reply:
Is that really the way you want to use a cli tool?  All my CLI stuff I did with `System.out` and only left logs on for `ERROR` (and higher) in case the signed state file stuff had problems.  As a tool _user_ the result was more what I wanted to see.

Reply:
This is used for command line tool. I tried to use logger and didn't see the logged msg. Talked to @alittley. Platform uses System.out (err) too.